Sources/PubNub/PubNub.swift

```diff
@@ -220,6 +220,59 @@ extension PubNub {
       completion?(result.map { $0.payload })
     }
   }
+
+  /// Publish a message to a channel and then adds a message action
+  ///
+  /// Message storage and TTL can be configured with the following rules:
+  /// 1. If `shouldStore` is true and `storeTTL` is 0, the message is stored with no expiry time.
+  /// 2. If `shouldStore` is true and `storeTTL` is X; X>0, the message is stored with an expiry time of X hours.
+  /// 3. If `shouldStore` is false or not specified, the message is not stored and the `storeTTL` parameter is ignored.
+  /// 4. If `storeTTL` is not specified, then expiration of the message defaults back to the expiry value for the key.
+  ///
+  /// - Parameters:
+  ///   - channel: The destination of the message
+  ///   - message: The message to publish
+  ///   - messageAction: The action added to the published message
+  ///   - shouldStore: If true the published message is stored in history.
+  ///   - storeTTL: Set a per message time to live in storage.
+  ///   - meta: Publish extra metadata with the request.
+  ///   - shouldCompress: Whether the message needs to be compressed before transmission
+  ///   - with: Additional network configuration to use on the request
+  ///   - respondOn: The queue the completion handler should be returned on
+  ///   - completion: The async result of the method call
+  public func publishWithMessageAction(
```

Comment:
This seems handy, why would it not just use the usual `publish` method with a parameter for the message action though?

Reply:
I wasn't sure if there was a practical use-case for this or not, but I needed something like this for integration tests 🤷‍♂

Reply:
The error being returned would specify the endpoint that was impacted, but if there is not a use-case for it then it can be moved to a test only extension.

Reply:
Yea that's a good point. I'm not sure we actually need this function.